fetch.bs

```diff
@@ -2049,6 +2051,29 @@ clearly stipulates that <a>connections</a> are keyed on
 <!-- See https://github.com/whatwg/fetch/issues/114#issuecomment-143500095 for when we make
      WebSocket saner -->
 
+<h3 id=http-cache-partition>HTTP Cache partition</h3>
+
+<p>To determine the associated <dfn export id=http-cache>HTTP cache</dfn> partition, given
+<var>request</var>, run these steps:
+
+<ol>
+ <li>
+  <p>If <var>request</var>'s <a for=request>client</a> is null, terminate these steps.</li>
+
+ <li>
+  <p>Let <var>environmentSettings</var> be <var>request</var>'s <a for=request>client</a>.</li>
+
+ <li>
+  <p>Let <var>topLevelOrigin</var> be <var>environmentSettings</var>'s top-level origin and
+  <var>origin</var> be <var>environmentSettings</var>'s origin.</li>
+
+ <li>
+  <p>Return the HTTP cache associated with <var>topLevelOrigin</var> and optionally,
+  <var>origin</var>.</li>
+</ol>
+
+<p class="note no-backref">User agents can use the registrable domains associated with
+<var>topLevelOrigin</var> and <var>origin</var> instead of the origins themselves.
```

Comment:
Hmm, I don't like how many degrees of freedom this gives us. It seems like there are now six configurations: (top-level origin + origin, top-level origin) x (registrable domain instead of top-level origin, top-level origin as-is) x (registrable domain instead of origin, origin as-is). Do we expect browsers to implement all six possible points on this matrix? That seems like a pretty bad failure of the consensus process. Can we get agreement on just one, or maybe two?

Reply:
Do you think we could leave the note regarding registrable domain altogether and let the spec just say "Return the HTTP cache associated with the <var>topLevelOrigin</var>, and optionally <var>origin</var>". In other words, is it ok for the spec to not be specific enough to mention registrable domain detail?

/cc @annevk : Does Firefox plan to use origin or registrable domain?

Reply:
So the proposal would be to reduce it to two possibilities? That's still not as good as one, but it'd be an improvement over six.

Reply:
We are reconsidering "site" (aka "agent cluster key" https://html.spec.whatwg.org/#obtain-agent-cluster-key) based on Chrome's feedback. Additionally Chrome's triple-keying made us reconsider omni-keying (each parent up to the top-level adds a key), but it's still early days in proving it's feasible.

Reply:
Since Chrome and Firefox are considering "site" and Safari already does that, changed it to use "agent cluster key". Although I could not get it to link to https://html.spec.whatwg.org/#obtain-agent-cluster-key. Would that require change in the way obtain-agent-cluster-key is defined in the html spec? Sorry, I am not very familiar with spec language.

Reply:
Added triple keying as the default option here.